### PR TITLE
refactor: Blue green example

### PR DESCRIPTION
**Pull request by allamand**
### What does this PR do?

This PR adapt the blue-green-upgrade example to uses external teams as of new blueprint direction https://github.com/aws-ia/terraform-aws-eks-blueprints/issues/1421

The example still rely on the internal addon module, as the new addon module don't actually allow us to create addons or workloads from terraform, A separate PR will update this part.

### More

- [x] Yes, I have tested the PR using my local account setup (Provide any test evidence report under Additional Notes)
- [x] Yes, I have updated the [docs](https://github.com/aws-ia/terraform-aws-eks-blueprints/tree/main/docs) for this feature
- [x] Yes, I ran `pre-commit run -a` with this PR

### For Moderators

- [ ] E2E Test successfully complete before merge?

### CODE REVIEW

File: examples/blue-green-upgrade/README.md [hunk 1]
```diff
@@ -41,7 +41,7 @@ See the Architecture of what we are building
 
 Our sample is composed of four main directory:
 
-- **core-infra** → this stack will create vpc and dependencies: create a Route53 sub zone for our sample, and a wildcard Certificate Manager certificate for our applications TLS endpoints, and a SecretManager password for the ArgoCD UIs.
+- **environment** → this stack will create the common vpc and dependencies used by our clusters: create a Route53 sub zone for our sample, and a wildcard Certificate Manager certificate for our applications TLS endpoints, and a SecretManager password for the ArgoCD UIs.
```
Comment (rodrigobersa) [inline examples/blue-green-upgrade/README.md:44]: I would rephrase this.
this stack will create the common VPC and its dependencies used by our EKS clusters: create a Route53 hosted zone for our sample, a wildcard certificate on Certificate Manager for our applications TLS endpoints, and a SecretManager password for the ArgoCD UIs.

File: examples/blue-green-upgrade/eks-green/main.tf [hunk 2]
```diff
@@ -4,22 +4,22 @@ provider "aws" {
 
 provider "kubernetes" {
   host                   = module.eks_cluster.eks_cluster_endpoint
-  cluster_ca_certificate = base64decode(module.eks_cluster.eks_cluster_certificate_authority_data)
+  cluster_ca_certificate = base64decode(module.eks_cluster.cluster_certificate_authority_data)
   token                  = data.aws_eks_cluster_auth.this.token
```
Comment (rodrigobersa) [inline examples/blue-green-upgrade/eks-green/main.tf:8]: Same for replacing the token.

File: examples/blue-green-upgrade/eks-green/main.tf [hunk 3]
```diff
@@ -31,14 +31,15 @@ data "aws_eks_cluster_auth" "this" {
 module "eks_cluster" {
   source = "../modules/eks_cluster"
 
-  suffix_stack_name = "green"
-  cluster_version   = "1.24" # Here, we deploy the cluster with the N+1 Kubernetes Version
+  aws_region      = var.aws_region
+  service_name    = "green"
+  cluster_version = "1.24" # Here, we deploy the cluster with the N+1 Kubernetes Version
```
Comment (rodrigobersa) [inline examples/blue-green-upgrade/eks-green/main.tf:36]: update to 1.26

```
cluster_version = "1.26"
```

File: examples/blue-green-upgrade/modules/eks_cluster/main.tf [hunk 4]
```diff
@@ -172,200 +256,388 @@ data "aws_caller_identity" "current" {}
...
+  source  = "aws-ia/eks-blueprints-teams/aws"
+  version = "~> 0.2"
+
+  name = "admin-team"
+
+  enable_admin = true
+  users = [
+    data.aws_caller_identity.current.arn,
+    "arn:${data.aws_partition.current.partition}:iam::${data.aws_caller_identity.current.account_id}:user/${local.iam_platform_user}",
+    "arn:aws:iam::${data.aws_caller_identity.current.account_id}:role/${local.eks_admin_role_name}"
```
Comment (rodrigobersa) [inline examples/blue-green-upgrade/modules/eks_cluster/main.tf:375]: Are we creating this user?

Add the `data_source` for partition.
```  "arn:${data.aws_partition.current.partition}:iam::${data.aws_caller_identity.current.account_id}:role/${local.eks_admin_role_name}"
```

File: examples/blue-green-upgrade/modules/eks_cluster/variables.tf [hunk 5]
```diff
@@ -13,7 +18,7 @@ variable "suffix_stack_name" {
 variable "cluster_version" {
   description = "The Version of Kubernetes to deploy"
   type        = string
-  default     = "1.23"
+  default     = "1.24"
```
Comment (rodrigobersa) [inline examples/blue-green-upgrade/modules/eks_cluster/variables.tf:21]: strike the default variable.

File: examples/blue-green-upgrade/README.md
Comment (rodrigobersa) [inline examples/blue-green-upgrade/README.md:1]: Provide the context the user should assume to access the cluster on line 161.

**Review by rodrigobersa — CHANGES_REQUESTED** [review]

File: examples/blue-green-upgrade/README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (allamand) [reply examples/blue-green-upgrade/README.md:44]: corrected, thanks

File: examples/blue-green-upgrade/environment/variables.tf
[the remark this answers is not in the document]
Comment (allamand) [reply examples/blue-green-upgrade/environment/variables.tf:1]: good catch

File: examples/blue-green-upgrade/eks-blue/main.tf [hunk 6]
```diff
@@ -31,14 +31,15 @@ data "aws_eks_cluster_auth" "this" {
 module "eks_cluster" {
   source = "../modules/eks_cluster"
 
-  suffix_stack_name = "blue"
-  cluster_version   = "1.23"
+  aws_region      = var.aws_region
+  service_name    = "blue"
+  cluster_version = "1.23"
```
[the remark this answers is not in the document]
Comment (allamand) [reply examples/blue-green-upgrade/eks-blue/main.tf:36]: I didn't test it yes in 1.25 will do to see if everything works

File: examples/blue-green-upgrade/eks-blue/main.tf [hunk 7]
```diff
@@ -4,22 +4,22 @@ provider "aws" {
 
 provider "kubernetes" {
   host                   = module.eks_cluster.eks_cluster_endpoint
-  cluster_ca_certificate = base64decode(module.eks_cluster.eks_cluster_certificate_authority_data)
+  cluster_ca_certificate = base64decode(module.eks_cluster.cluster_certificate_authority_data)
   token                  = data.aws_eks_cluster_auth.this.token
```
[the remark this answers is not in the document]
Comment (allamand) [reply examples/blue-green-upgrade/eks-blue/main.tf:8]: many example are using the static provisioning as per the FAQ say: https://github.com/aws-ia/terraform-aws-eks-blueprints/blob/main/FAQ.md?plain=1 so wonder why change it here ?

File: examples/blue-green-upgrade/eks-green/main.tf (hunk 3 above)
Comment (allamand) [reply examples/blue-green-upgrade/eks-green/main.tf:36]: i'll test it and update

File: examples/blue-green-upgrade/modules/eks_cluster/main.tf (hunk 4 above)
Comment (allamand) [reply examples/blue-green-upgrade/modules/eks_cluster/main.tf:375]: We don't create the platform_user, in the terraform.tfvars we can specify the name of the user to be added to the platform team.

What do you mean by add the data_source for partition ?

File: examples/blue-green-upgrade/modules/eks_cluster/variables.tf (hunk 5 above)
Comment (allamand) [reply examples/blue-green-upgrade/modules/eks_cluster/variables.tf:21]: 1.25

File: examples/blue-green-upgrade/README.md
Comment (allamand) [reply examples/blue-green-upgrade/README.md:1]: ok

File: examples/blue-green-upgrade/eks-blue/main.tf (hunk 6 above)
[the remark this answers is not in the document]
Comment (allamand) [reply examples/blue-green-upgrade/eks-blue/main.tf]: done

File: examples/blue-green-upgrade/environment/main.tf [hunk 8]
```diff
@@ -43,6 +53,17 @@ module "vpc" {
...
+  #     # Ignore changes to tags, e.g. because a management agent
+  #     # updates these based on some ruleset managed elsewhere.
+  #     public_subnet_tags,
+  #     #private_subnet_tags,
+  #   ]
+  # }
+
 }
 
 # Retrieve existing root hosted zone
```
[the remark this answers is not in the document]
Comment (allamand) [reply examples/blue-green-upgrade/environment/main.tf:69]: let's keep public for now and we'll test if it can also work in private

**Comment by allamand** [comment]
@rodrigobersa thanks for the review, i think we are good now

File: examples/blue-green-upgrade/modules/eks_cluster/main.tf [hunk 9]
```diff
@@ -1,33 +1,63 @@
...
   route53_weight             = var.route53_weight
   ecsfrontend_route53_weight = var.ecsfrontend_route53_weight
 
-  tag_val_vpc            = var.vpc_tag_value == "" ? var.core_stack_name : var.vpc_tag_value
-  tag_val_private_subnet = var.vpc_tag_value == "" ? "${var.core_stack_name}-private-" : var.vpc_tag_value
+  eks_cluster_domain = "${local.environment}.${local.hosted_zone_name}" # for external-dns
+
+  tag_val_vpc            = local.environment
+  tag_val_public_subnet  = "${local.environment}-public-"
+  tag_val_private_subnet = "${local.environment}-private-"
```
[the remark this answers is not in the document]
Comment (allamand) [reply examples/blue-green-upgrade/modules/eks_cluster/main.tf:47]: yes, used to find them by tag name

File: examples/blue-green-upgrade/modules/eks_cluster/main.tf [hunk 10]
```diff
@@ -455,6 +707,187 @@ module "kubernetes_addons" {
...
+#         #domainFilters = [data.aws_route53_zone.sub.zone_id]
+#         domainFilters = [local.eks_cluster_domain]
+#       })
+#     ]
+#   }
+
+#   #enable_kubecost = true
+#   enable_argo_rollouts = local.argo_rollouts
+
+# }
```
[the remark this answers is not in the document]
Comment (allamand) [reply examples/blue-green-upgrade/modules/eks_cluster/main.tf:841]: This bloc was the usage of v5 addon module, but I can put it elsewhere for the moment (as gitops bridge is still necessary and does not exist in v5 addon for now)

File: examples/blue-green-upgrade/modules/eks_cluster/main.tf [hunk 11]
```diff
@@ -71,21 +101,75 @@ locals {
...
+              }
+            ]
+          }
+
           ecsdemoFrontend = {
-            repoURL        = "https://github.com/aws-containers/ecsdemo-frontend"
+            repoURL        = "https://github.com/allamand/ecsdemo-frontend"
             targetRevision = "main"
-            replicaCount   = "3"
+            #replicaCount   = "9" # see autoscaling configuration
```
[the remark this answers is not in the document]
Comment (allamand) [reply examples/blue-green-upgrade/modules/eks_cluster/main.tf:172]: I was wondering If I keep this to teach users that replicaCount is in fact managed dynamically in the autoscaling configuration. I removed if you feel it is unnecessary

File: examples/blue-green-upgrade/modules/eks_cluster/outputs.tf [hunk 12]
```diff
@@ -1,19 +1,34 @@
...
+  value       = module.eks.cluster_name
 }
 
-output "configure_kubectl" {
-  description = "Configure kubectl: make sure you're logged in with the correct AWS profile and run the following command to update your kubeconfig"
-  value       = module.eks_blueprints.configure_kubectl
+# output "configure_kubectl" {
+#   description = "Configure kubectl: make sure you're logged in with the correct AWS profile and run the following command to update your kubeconfig"
+#   value       = module.eks.configure_kubectl
+# }
```
[the remark this answers is not in the document]
Comment (allamand) [reply examples/blue-green-upgrade/modules/eks_cluster/outputs.tf:9]: fixed

File: examples/blue-green-upgrade/modules/eks_cluster/main.tf (hunk 10 above)
[the remark this answers is not in the document]
Comment (rodrigobersa) [reply examples/blue-green-upgrade/modules/eks_cluster/main.tf]: It's better to keep the code clean. Keep it in your branches for future enhancement.

**Review by rodrigobersa — COMMENTED** [review]
2 minor comments and we're good to go!

**Review by rodrigobersa — APPROVED** [review]
